vivisect/extensions/__init__.py

```diff
@@ -32,23 +33,26 @@ def loadExtensions(vw, vwgui):
             modpath = os.path.join(dirname, fname)
             # dig into first level of directories and exec the __init__.py
             if os.path.isdir(modpath):
+                modname = fname
                 modpath = os.path.join(modpath, '__init__.py')
                 if not os.path.exists(modpath):
                     continue
 
             # otherwise, run all the .py files in the VIV_EXT_PATH dir
-            if not modpath.endswith('.py'):
+            elif not modpath.endswith('.py'):
                 continue
 
-            # Build code objects from the module files
-            spec = importlib.util.spec_from_file_location('viv_ext', modpath)
-            module = importlib.util.module_from_spec(spec)
-            spec.loader.exec_module(module)
+            else:
+                # it's a .py file
+                modname = fname[:-3]
 
             try:
-                with open(modpath, 'r') as fd:
-                    filebytes = fd.read()
-                exec(filebytes, module.__dict__)
+                # Build code objects from the module files
+                spec = importlib.util.spec_from_file_location(fname, modpath)
+                module = importlib.util.module_from_spec(spec)
+                module.vw = vw
```

Comment:
I'm not against it, but is this something we want to necessarily punch in, given that when we call module.vivExtension we pass that in anyway?